Adapters/Vungle/ISVungleAdapter/ISVungleAdapter.m

```diff
@@ -341,22 +255,16 @@ - (void)initAndLoadRewardedVideoWithUserId:(NSString *)userId
     }
     
     LogInternal_Internal(@"appId = %@, placementId = %@", appId, placementId);
-    
-    // add to rewarded video delegate map
-    [_rewardedVideoPlacementIdToSmashDelegate setObject:delegate
-                                                 forKey:placementId];
-    
-    // add rewarded video to singleton
-    [[ISVungleAdapterSingleton sharedInstance] addRewardedVideoDelegate:self
-                                                                 forKey:placementId];
-    
+
+    self.rewardedVideoAdapterRouter = [[ISVungleRewardedVideoAdapterRouter alloc] initWithPlacementID:placementId parentAdapter:self delegate:delegate];
+
     switch (_initState) {
         case INIT_STATE_NONE:
         case INIT_STATE_IN_PROGRESS:
             [self initSDKWithAppId:appId];
             break;
         case INIT_STATE_SUCCESS:
-            [self loadRewardedVideoInternalWithPlacement:placementId];
+            [self loadRewardedVideoInternalWithPlacement];
             break;
         case INIT_STATE_FAILED: {
             LogAdapterApi_Internal(@"init failed - placementId = %@", placementId);
```

Comment:
Should we also nil the router reference here. just to be more safe??

Reply:
@ManojBudumuru-vungle Updated, thanks a lot. :)